src/BaGet.Core/Mirror/Clients/MirrorV2Client.cs

```diff
@@ -0,0 +1,114 @@
+using BaGet.Protocol.Models;
+using Microsoft.Extensions.Options;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace BaGet.Core
+{
+    internal sealed class MirrorV2Client : IMirrorNuGetClient
+    {
+        private readonly ILogger _logger;
+        private readonly SourceCacheContext _cache;
+        private readonly SourceRepository _repository;
+
+        public MirrorV2Client(IOptionsSnapshot<MirrorOptions> options)
+        {
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (options.Value?.PackageSource?.AbsolutePath == null)
+            {
+                throw new ArgumentException("No mirror package source has been set.");
```

Comment:
Ideally this should be done at startup by ASP.NET Core's options validation. See my comment up above: https://github.com/loic-sharma/BaGet/pull/630/files#r582447197

If that idea works, you should be able to remove this check.